### PR TITLE
Fixing issue with folder names and cleaning up CS errors

### PR DESCRIPTION
**Pull request by matdave**
### What does it do?
Allows special characters in folder names.

### Why is it needed?
ExtJs was having an issue when utilizing special characters, causing the user to be unable to edit a file or folder. 

Additionally added some logic optimization suggestions from CodeSniffer

### How to test
I created a folder named `umlaute (äöü)` and was unable to interact with it until applying the changes. 

### Related issue(s)/PR(s)
Issue #15659

### CODE REVIEW

**Comment by Ibochkarev** [comment]
@cla-bot check

**Comment by Mark-H** [comment]
Given the actual change is a few lines vs hundreds of lines of code style changes.... not doing that together would've been much more useful ;)

**Review by theboxer — APPROVED** [review]

**Review by Ibochkarev — APPROVED** [review]
Tested. This fix solves the specified problem

**Comment by opengeek** [comment]
@matdave — has this also been addressed for 3.x?

**Comment by matdave** [comment]
> @matdave — has this also been addressed for 3.x?

yes

**Comment by opengeek** [comment]
> > @matdave — has this also been addressed for 3.x?
> 
> yes

Ummm, could you be more specific as to where and when this has been addressed for 3.x? Sigh…

**Comment by matdave** [comment]
> > > @matdave — has this also been addressed for 3.x?
> > 
> > 
> > yes
> 
> Ummm, could you be more specific as to where and when this has been addressed for 3.x? Sigh…

3.x uses the FlySystem, I'm not sure exactly when but at some point someone added the same `rawurlencode` around the "id" parameter. That is what I ported to 2.x in this update.

File: core/model/modx/sources/mods3mediasource.class.php [hunk 1]
```diff
@@ -1292,7 +1372,7 @@ public function getObjectContents($objectPath) {
             'last_accessed' => '',
             'last_modified' => '',
             'content' => $contents,
-            'image' => in_array($fileExtension,$imageExtensions) ? true : false,
+            'image' => in_array($fileExtension, $imageExtensions, true),
```
Comment (opengeek) [inline core/model/modx/sources/mods3mediasource.class.php:1375]: This is a non-trivial change.

**Review by opengeek — CHANGES_REQUESTED** [review]
There seem to be too many changes besides the folder name here for me to accept this as is. If you could, please undo everything except the folder name change and update the PR.

**Comment by JoshuaLuckers** [comment]
@matdave would it be possible for you to address the change request?

**Comment by rthrash** [comment]
Mat won't be back until next week, FYI.

**Pull request closed without merging**